measurement/plugins/netflix_fast/measurements.py

```diff
@@ -0,0 +1,334 @@
+
+        completed_time = time.time()
+        elapsed_time = completed_time - start_time
+
+        # If this is the first thread to complete, record the time and total at this point
+        if self.completed_elapsed_time is None:
+            self.completed_elapsed_time = elapsed_time
+            for global_thread_result in self.thread_results:
+                self.completed_total += global_thread_result["download_size"]
+
+        thread_result["download_rate"] = (
+            thread_result["download_size"] / elapsed_time / 1024 / 1024 * 8
+        )
+        thread_result["elapsed_time"] = elapsed_time
+        self.finished_threads += 1
+
+    def _query_api(self, s, token):
+        params = {"https": "true", "token": token, "urlCount": self.urlcount}
+        # '/v2/' path returns all location data about the servers
+        api_resp = s.get("https://api.fast.com/netflix/speedtest/v2", params=params)
+        api_json = api_resp.json()
+        for i in range(len(api_json["targets"])):
+            self.thread_results[i]["url"] = api_json["targets"][i]["url"]
+            self.thread_results[i]["location"] = api_json["targets"][i]["location"]
+        self.client_data = api_json["client"]
+        return
+
+    def _is_stabilised(self, percent_deltas, elapsed_time):
+        return (
+            elapsed_time > MIN_TIME
+            and len(percent_deltas) >= STABLE_MEASUREMENTS_LENGTH
+            and max(percent_deltas) < STABLE_MEASUREMENTS_DELTA
+        )
+
+    def _get_response(self, s):
+        return s.get("http://fast.com/")
+
+    def _get_connection(self, url):
+        s = requests.Session()
+        self.sessions.append(s)
+        conn = s.get(url, stream=True)
+        return conn
+
+    def _is_test_complete(self, elapsed_time, percent_deltas):
+        if elapsed_time > self.max_time:
+            return "time_expired"
+        if (self.terminate_on_result_stable) & (
+            self._is_stabilised(percent_deltas, elapsed_time)
+        ):
+            return "result_stabilised"
+        if self.finished_threads == len(self.thread_results):
+            return "all_complete"
+        if (self.terminate_on_thread_complete) & (self.finished_threads >= 1):
+            return "thread_complete"
+        return False
+
+    def _get_url_result(self, thread_result):
+        host = urllib.parse.urlparse(thread_result["url"]).netloc
+        city = thread_result["location"]["city"]
+        country = thread_result["location"]["country"]
+        LatencyResult = LatencyMeasurement(self.id, host, count=PING_COUNT).measure()[0]
+
+        return NetflixFastThreadResult(
+            id=self.id,
+            host=host,
+            city=city,
+            country=country,
+            download_size=thread_result["download_size"],
+            download_size_unit=StorageUnit("B"),
+            download_rate=thread_result["download_rate"],
+            download_rate_unit=NetworkUnit("Mbit/s"),
```

Comment:
Can we just keep all of our units internal to the measurement in `bit/s` (as alluded to by @sjdines above) and then just use `bit/s` here?